src/app/pages/questionnaire-page/questionnaire-page.component.ts

```diff
@@ -80,17 +83,22 @@ export class QuestionnairePageComponent implements OnInit {
   submitQuestionnaire() {
     this.submitting = true;
 
+    this.genderError = false;
```

Comment:
The questionnaire is not submitted if genderError = false; or decimalError = false; and presented with the error which means we get the error after if they didn't choose any option or inputted decimal number or anything other than numbers in the 'age in months' field.